### PR TITLE
[JENKINS-46358] Whitelist new StringGroovyMethods

### PR DESCRIPTION
**Pull request by abayer**
[JENKINS-46358](https://issues.jenkins-ci.org/browse/JENKINS-46358)

These will be added to CPS in https://github.com/cloudbees/groovy-cps/pull/70

cc @reviewbybees

### CODE REVIEW

**Comment by svanoort** [comment]
Looks like the sanity check is failing -- something about methods not existing?  I'm assuming that's an issue with the sanity check itself and the new rules are fine though

**Comment by abayer** [comment]
@svanoort Nah, they were legit sanity failures, which I've now cleaned up.

File: src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptor.java [hunk 1]
```diff
@@ -93,17 +95,25 @@
...
-                    } else {
-                        throw StaticWhitelist.rejectStaticMethod(dgmMethod);
+                    } else if (foundDgmMethod == null) {
+                        foundDgmMethod = dgmMethod;
                     }
                 }
             }
 
+            // Some methods are found by GroovyCallSiteSelector in both DefaultGroovyMethods and StringGroovyMethods, so
+            // we're iterating over the whole list before we decide to fail out on the first failure we found.
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptor.java]: Makes sense, I think, but would be nice to have a test case proving that this change is effective.

Comment (abayer) [inline src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptor.java]: Good point. I've got that tested over in `workflow-CPS` but not here.

**Pull request merged**